src/lib/demux.rs

```diff
@@ -457,6 +470,30 @@ where
                 None => (),
             }
 
+            // If desired, get the sample barcode bases from the FASTQ header (of the first read)
+            if self.sample_barcode_in_fastq_header {
+                // get the FASTQ header from the first read
+                let read = &zipped_reads[0];
+                let header =
+                    FastqHeader::try_from(read.head()).with_context(|| {
```

Comment:
As an aside, I notice that we call `FastqHeader::try_from` in a number of places, but do not cache its value.  That could be an area ripe for speed improvements.  If you agree, I can track as an issue.

Reply:
Probably worthwhile to try ... although I think it's not that expensive as it's almost all wrappers around slices until you modify it.  You might also run into lifetime issues, and if you do I'd give up quickly.

Reply:
Deferring, as it's only adding one more.